miscelanea/003-scc-and-caps/README.md

```diff
@@ -0,0 +1,492 @@
+the systemd container interface document.
+
+### tar: var/lib/chrony: Cannot change ownership to uid 285, gid 283: Operation not permitted
+
+Fixed by adding the `CHOWN` capability.
+
+```raw
+Make arbitrary changes to file UIDs and GIDs (see chown(2)).
+```
+
+### tar: var/lib/chrony: Cannot change mode to rwxr-xr-x: Operation not permitted
+
+Fixed by adding the `FOWNER` capability.
+
+```raw
+* Bypass permission checks on operations that normally require the filesystem UID of the process to match the UID of the file (e.g., chmod(2), utime(2)), excluding  those  operations  covered  by  CAP_DAC_OVERRIDE  and  CAP_DAC_READ_SEARCH;
+* set inode flags (see ioctl_iflags(2)) on arbitrary files;
+* set Access Control Lists (ACLs) on arbitrary files;
+* ignore directory sticky bit on file deletion;
+* modify user extended attributes on sticky directory owned by any user;
+* specify O_NOATIME for arbitrary files in open(2) and fcntl(2).
+```
+
+### rm: cannot remove '/run/rpcbind/*': Permission denied
+
+Fixed by adding the `DAC_OVERRIDE` capability.
+
+```raw
+Bypass file read, write, and execute permission checks.  (DAC is an abbreviation of "discretionary access control".)
+```
+
+### PR_SET_MM_ARG_START failed, proceeding without: Operation not permitted
+
+Error on initing systemd, at the very beginning.
+
+Fixed by adding the `SYS_RESOURCE` capability.
+
+```raw
+* Use reserved space on ext2 filesystems;
+* make ioctl(2) calls controlling ext3 journaling;
+* override disk quota limits;
+* increase resource limits (see setrlimit(2));
+* override RLIMIT_NPROC resource limit;
+* override maximum number of consoles on console allocation;
+* override maximum number of keymaps;
+* allow more than 64hz interrupts from the real-time clock;
+* raise msg_qbytes limit for a System V message queue above the limit in /proc/sys/kernel/msgmnb (see msgop(2) and msgctl(2));
+* allow the RLIMIT_NOFILE resource limit on the number of "in-flight" file descriptors to be bypassed when passing file descriptors to another process via a UNIX domain socket (see unix(7));
+* override the /proc/sys/fs/pipe-size-max limit when setting the capacity of a pipe using the F_SETPIPE_SZ fcntl(2) command.
+* use F_SETPIPE_SZ to increase the capacity of a pipe above the limit specified by /proc/sys/fs/pipe-max-size;
+* override /proc/sys/fs/mqueue/queues_max limit when creating POSIX message queues (see mq_overview(7));
+* employ the prctl(2) PR_SET_MM operation;
+```
+
+### systemd-tmpfiles-setup.service: Failed to set invocation ID on control group /kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-podb5886319_457f_4071_b775_834cd8b166f0.slice/crio-11b9d211f76f0b2d0b4de4effd7126fcfac4c428359d2148671a783d11f3c803.scope/system.slice/systemd-tmpfiles-setup.service, ignoring: Operation not permitted
+
+Fixed by adding the `SYS_ADMIN`. This is the big one to try to remove in
+a second stage as this provide too many privileges.
```

Comment:
So it looks like something related to cgroups v1?

Any idea which of the *many* capabilities provided by `CAP_SYS_ADMIN` are the critical ones here?

Reply:
I would say it is related with both cgroup v1 and v2. I have investigated deeper, and I can not see different code for cgroup v1 or v2 on this. However I can see an early assignment of `SYS_ADMIN` capability. Thanks to your comment I have seen that, and this is fixed with just adding `SETFCAP` capability. The above error message is happening at this point:

https://github.com/systemd/systemd/blob/69c0807432fa4fbfbf507a53872664cd26715559/src/core/cgroup.c#L613

Following the breadcrumbs I find that `setxattr` need the capability `SETFCAP`. (man capabilities, and search setxattr).

About remove the `SYS_ADMIN`, the following links could be helpful to remove it, but it requires another investigation. It is a pending task that I would like to do. The specific documents are:
- https://systemd.io/CONTAINER_INTERFACE/
- https://myles.sh/configuring-lxc-unprivileged-containers-in-debian-jessie/
- https://blog.iwakd.de/lxc-cap_sys_admin-jessie

Reply:
About the second question, I would need to go deeper. No idea currently.

Reply:
No worries, thanks for the extra info.